packages/in_app_purchase/lib/src/in_app_purchase_connection/google_play_connection.dart

```diff
@@ -34,6 +34,51 @@ class GooglePlayConnection
     return _billingClient.isReady();
   }
 
+  @override
+  Future<QueryPastPurchaseResponse> queryPastPurchases(
+      {String applicationUserName}) async {
+    List<PurchasesResultWrapper> responses = await Future.wait([
+      _billingClient.queryPurchaseHistory(SkuType.inapp),
+      _billingClient.queryPurchaseHistory(SkuType.subs)
+    ]);
+
+    BillingResponse errorInApp = responses.first.responseCode;
+    BillingResponse errorSubs = responses.last.responseCode;
+    String errorMessage = null;
+    if (errorInApp != BillingResponse.ok) {
+      errorMessage = errorInApp.toString();
+    }
+    if (errorSubs != BillingResponse.ok && errorSubs != errorInApp) {
+      errorMessage += ', ${errorSubs.toString()}';
+    }
+
+    List<PurchaseDetails> pastPurchases;
+    if (responses == null) {
+      pastPurchases = [];
+    } else {
+      pastPurchases = responses.expand((PurchasesResultWrapper response) {
+        return response.purchasesList;
+      }).map((PurchaseWrapper purchaseWrapper) {
+        return purchaseWrapper.toPurchaseDetails();
+      }).toList();
+    }
+    return QueryPastPurchaseResponse(
+      pastPurchases: pastPurchases,
+      error: errorMessage != null
+          ? PurchaseError(
+              source: PurchaseSource.GooglePlay,
+              code: 'restore_transactions_failed',
+              message: {'message': errorMessage})
+          : null,
+    );
+  }
+
+  @override
+  Future<PurchaseVerificationData> refreshPurchaseVerificationData(
```

Comment:
I think it's worth documenting on this that it's a non-op since there's no refreshing this data from Play.